From a744b37802d198af2a266ecedaff365a661bb56e Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Wed, 24 Apr 2024 18:38:06 +0200
Subject: [PATCH] net: fec_mxc: add weak implementation of board_get_mac

On other imx8 based platforms board_get_mac may not be implemented and
cause linker-errors. Add weak implementation to fix the build.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 drivers/net/fec_mxc.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/fec_mxc.c b/drivers/net/fec_mxc.c
index f2681a2e5c9..73e4fc642a2 100644
--- a/drivers/net/fec_mxc.c
+++ b/drivers/net/fec_mxc.c
@@ -394,6 +394,11 @@ static void fec_rbd_clean(int last, struct fec_bd *prbd)
 	writew(0, &prbd->data_length);
 }
 
+__weak int board_get_mac(int dev_id, unsigned char *mac)
+{
+	return -EOPNOTSUPP;
+};
+
 static int fec_get_hwaddr(int dev_id, unsigned char *mac)
 {
 	board_get_mac(dev_id, mac);
